File: SimpleHabit/app/src/main/java/com/me/simplehabit/HabitApplication.java
package com.me.simplehabit;

import android.app.Application;

import com.me.simplehabit.data.models.CategoryProgramModelImpl;
import com.me.simplehabit.data.models.CurrentProgramModelImpl;
import com.me.simplehabit.data.models.TopicModelImpl;

public class HabitApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        CurrentProgramModelImpl.initCurrentProgramModel(getApplicationContext());
        TopicModelImpl.initTopicModel(getApplicationContext());
        CategoryProgramModelImpl.initCategoryProgramModel(getApplicationContext());
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/HabitApplication.kt
package com.me.simplehabit

import android.app.Application

import com.me.simplehabit.data.models.CategoryProgramModelImpl
import com.me.simplehabit.data.models.CurrentProgramModelImpl
import com.me.simplehabit.data.models.TopicModelImpl

class HabitApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        CurrentProgramModelImpl.initDatabase(applicationContext)
        TopicModelImpl.initDatabase(applicationContext)
        CategoryProgramModelImpl.initDatabase(applicationContext)
    }
}
